pkg/clusteragent/admission/mutate/auto_instru.go

```diff
@@ -0,0 +1,175 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build kubeapiserver
+// +build kubeapiserver
+
+package mutate
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/DataDog/datadog-agent/pkg/clusteragent/admission/metrics"
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/dynamic"
+)
+
+const (
+	// Shared config
+	initContainerName = "datadog-tracer-init"
+	volumeName        = "datadog-auto-instrumentation"
+	mountPath         = "/datadog"
+
+	// Java config
+	javaToolOptionsKey   = "JAVA_TOOL_OPTIONS"
+	javaToolOptionsValue = " -javaagent:/datadog/dd-java-agent.jar"
+)
+
+var (
+	tracerVersionAnnotationKeyFormat = "admission.datadoghq.com/%s-tracer.version"
+	customTracerAnnotationKeyFormat  = "admission.datadoghq.com/%s-tracer.custom-image"
```

Comment:
This one is mostly for testing, doesn't have to be publicly documented. We can drop support once we have the official library images.

Reply:
it can be useful for customers using Datadog without internet (container registry access). if they pull and push to their registry with another path than "dd-<lang>-agent-init".